### ISSUE TITLE
libc++.1.dylib (no such file)

### ISSUE THREAD

**Issue by SlistInc** [storizzi/Deep-Live-Cam-Mac-Installer#3]
I tried to run ./deep_live_cam.sh but keep running into the following error/issue
`Using Python from /opt/homebrew/Caskroom/miniconda/base/envs/deep-live-cam/bin/python: Python 3.10.14
Using pip from /opt/homebrew/Caskroom/miniconda/base/envs/deep-live-cam/bin/pip: pip 24.2 from /opt/homebrew/Caskroom/miniconda/base/envs/deep-live-cam/lib/python3.10/site-packages/pip (python 3.10)
Checking camera access...
Camera access is authorized.
Accessing the camera...
2024-08-13 10:22:35.751 swift-frontend[35334:483717] WARNING: AVCaptureDeviceTypeExternal is deprecated for Continuity Cameras. Please use AVCaptureDeviceTypeContinuityCamera and add NSCameraUseContinuityCameraDeviceType to your Info.plist.
Camera session started.
Running the application with CoreML acceleration...
Traceback (most recent call last):
  File "/Users/X/Local_Documents/Local_Projects/Deep-Live-Cam-Mac-Installer/Deep-Live-Cam/run.py", line 3, in <module>
    from modules import core
  File "/Users/X/Local_Documents/Local_Projects/Deep-Live-Cam-Mac-Installer/Deep-Live-Cam/modules/core.py", line 20, in <module>
    import modules.ui as ui
  File "/Users/X/Local_Documents/Local_Projects/Deep-Live-Cam-Mac-Installer/Deep-Live-Cam/modules/ui.py", line 10, in <module>
    from modules.face_analyser import get_one_face
  File "/Users/X/Local_Documents/Local_Projects/Deep-Live-Cam-Mac-Installer/Deep-Live-Cam/modules/face_analyser.py", line 2, in <module>
    import insightface
  File "/opt/homebrew/Caskroom/miniconda/base/envs/deep-live-cam/lib/python3.10/site-packages/insightface/__init__.py", line 18, in <module>
    from . import app
  File "/opt/homebrew/Caskroom/miniconda/base/envs/deep-live-cam/lib/python3.10/site-packages/insightface/app/__init__.py", line 2, in <module>
    from .mask_renderer import *
  File "/opt/homebrew/Caskroom/miniconda/base/envs/deep-live-cam/lib/python3.10/site-packages/insightface/app/mask_renderer.py", line 8, in <module>
    from ..thirdparty import face3d
  File "/opt/homebrew/Caskroom/miniconda/base/envs/deep-live-cam/lib/python3.10/site-packages/insightface/thirdparty/face3d/__init__.py", line 3, in <module>
    from . import mesh
  File "/opt/homebrew/Caskroom/miniconda/base/envs/deep-live-cam/lib/python3.10/site-packages/insightface/thirdparty/face3d/mesh/__init__.py", line 9, in <module>
    from .cython import mesh_core_cython
ImportError: dlopen(/opt/homebrew/Caskroom/miniconda/base/envs/deep-live-cam/lib/python3.10/site-packages/insightface/thirdparty/face3d/mesh/cython/mesh_core_cython.cpython-310-darwin.so, 0x0002): Library not loaded: @rpath/libc++.1.dylib
  Referenced from: <3CF8BEC7-7301-3ACC-90A6-CBC41134B979> /opt/homebrew/Caskroom/miniconda/base/envs/deep-live-cam/lib/python3.10/site-packages/insightface/thirdparty/face3d/mesh/cython/mesh_core_cython.cpython-310-darwin.so
  Reason: tried: '/opt/homebrew/Caskroom/miniforge/base/lib/libc++.1.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/opt/homebrew/Caskroom/miniforge/base/lib/libc++.1.dylib' (no such file), '/opt/homebrew/Caskroom/miniforge/base/lib/libc++.1.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/opt/homebrew/Caskroom/miniforge/base/lib/libc++.1.dylib' (no such file), '/opt/homebrew/Caskroom/miniconda/base/envs/deep-live-cam/bin/../lib/libc++.1.dylib' (no such file), '/opt/homebrew/Caskroom/miniconda/base/envs/deep-live-cam/bin/../lib/libc++.1.dylib' (no such file)`
  
  What I tried:
- Updating and upgrading brew
- Running ./deep_live_cam.sh --clean and retrying to run ./deep_live_cam.sh

My system:
- Apple M1 Max, 32 Gb
- macOS Sonoma 14.6.1 (23G93)

Maybe there is a simple fix, I hope someone can help me on this issue.

**Comment by storizzi** [comment]
Thanks for reporting this issue.

This looks like xcode developer tools might be missing on your system - I didn't realised this was a prerequisite, so I've updated the script. It will now check for whether xcode tools exists and try to download them if they are not present.

You can update the script (git pull from the project directory) and try re-running the script. I've added instructions for how to do this to the README as well.

Let me know if this fixes your issue - hopefully it will!

**Comment by SlistInc** [comment]
Thanks for the quick input. Unfortunately that was not the solution since I already have the developer tools installed.

`xcode-select --install
`

results in

`xcode-select: note: Command line tools are already installed. Use "Software Update" in System Settings or the softwareupdate command line interface to install updates`

and

`softwareupdate -i -a
`

results in

```
Software Update Tool

Finding available software
No updates are available.

```

There must be something else, I kept my system as updated as I could.

**Comment by storizzi** [comment]
Could you try `brew doctor` and follow any advice given then `brew cleanup` then

```
./deep_live_cam.sh --clean
git pull
./deep_live_cam.sh
```

**Comment by SlistInc** [comment]
**TLDR:** I solved it by copying a local libc++.1.dylib into '/opt/homebrew/Caskroom/miniconda/base/envs/deep-live-cam/bin/../lib/libc++.1.dylib'.

**Long:**
Thanks for the update. Still, the issue remained the same. I manage however to solve it by running

`sudo find / -name libc++.1.dylib`

and then locating some libc++.1.dylib files I had from other installs (mainly Pinokio). I copied this file to '/opt/homebrew/Caskroom/miniconda/base/envs/deep-live-cam/bin/../lib/libc++.1.dylib' and get no more error / the GUI opens without any hick. Thanks for your support!
